examples/pub-sub-presence/javascript/src/script.ts

```diff
@@ -0,0 +1,71 @@
+import * as Ably from 'ably';
+import { faker } from '@faker-js/faker';
+import './styles.css';
+
+const client = new Ably.Realtime({
+  clientId: faker.person.firstName(),
+  key: import.meta.env.VITE_PUBLIC_ABLY_KEY as string,
+});
+const channel = client.channels.get('viewer-presence');
+
+async function initializePresence() {
+  await channel.presence.subscribe((member) => {
+    const { action, clientId, data } = member;
+
+    switch (action) {
```

Comment:
I believe we don't recommend this, it can lead to some issues, it's better to refetch presence set here (we are going to provide presence set as one of listener's argument later to make it more useful). Action can be used to show toasters or other notification, not for rebuilding the set (because getting the initial set is not atomic)